api/lib/topic-lib.js

```diff
@@ -0,0 +1,65 @@
+var Topic = require('../model/topic');
+var Comment = require('../model/comment');
+
+module.exports.addToTopic = function addToTopicDB(options, callback){
+  var newTopic = new Topic({
+    topicQuestion: options.question,
+    tag: options.tag
+  });
+  newTopic.save(function (err) {
+    if (err) {
+      return callback(err);
+    }
+    // saved!
+    console.log('saved to topic');
+    return callback(null);
+  })
+}
+
+module.exports.getAllTopics = function getAllTopics(options, callback){
+  Topic.find({}, function(err, topics) {
+    if(err)
+      return callback(err);
+    return callback(null, topics);
+  });
+}
+
+module.exports.addTopic = function addTopic(options, callback){
+  var newTopic = new Topic({
+    topicQuestion: options.question,
+    tag: options.tag,
+    authorName: options.author
+  });
+  newTopic.save(function (err) {
+    if (err) {
+      return callback(err);
+    }
+    // saved!
+    console.log('saved to topic');
+    return callback(null);
```

Comment:
maybe make this return the `result` which could be the id of the newly created topic